src/robotFactory.js

```diff
@@ -1,10 +1,81 @@
 'use strict';
 
-class BaseRobot {}
+class BaseRobot {
+  constructor(name, weight, coords, chipVersion) {
+    this.name = name;
+    this.weight = weight;
 
-class FlyingRobot {}
+    this.coords = {
+      x: coords.x || 0,
+      y: coords.y || 0,
+    };
+    this.chipVersion = chipVersion;
+  }
 
-class DeliveryDrone {}
+  goForward(step = 1) {
+    this.coords.y += step;
+  }
```

Comment:
```suggestion
  goForward(step = 1) {
    this.coords.y += step;
    
    return this;
  }
```
If you add `return this` to movement functions, you can chain them.